src/server/index.tsx

```diff
@@ -52,12 +52,34 @@ server.get("/css/code-themes/:name", CodeThemeHandler);
 server.get("/css/themelist", ThemesListHandler);
 server.get("/*", CatchAllHandler);
 
-server.listen(Number(port), hostname, () => {
+var listener = server.listen(Number(port), hostname, () => {
   setupDateFns();
-  console.log(`http://${hostname}:${port}`);
+  console.log(`Started listening on http://${hostname}:${port}`);
 });
 
-process.on("SIGINT", () => {
-  console.info("Interrupted");
-  process.exit(0);
+var signals = {
+  'SIGHUP': 1,
+  'SIGINT': 2,
+  'SIGTERM': 15
+};
+
+const shutdown = (signal, value) => {
+  // TODO: Should set a flag here for the listener to reject any further
+  // incoming connections with a HTTP 503 error while shutting down.
+  // Otherwise the connection count may not drop to 0 before timeout.
+  listener.close(() => {
+    console.log(`Lemmy stopped by ${signal} with value ${value}`);
+    process.exit(128 + value);
```

Comment:
What does `128 + value` mean?

Reply:
Would be good to make that a const, and add a comment for it.